src/pages/workspace/accounting/qboConnectionButton/index.native.tsx

```diff
@@ -0,0 +1,71 @@
+import React, {useCallback, useRef, useState} from 'react';
+import {withOnyx} from 'react-native-onyx';
+import WebView from 'react-native-webview';
+import type {WebViewNavigation} from 'react-native-webview';
+import Button from '@components/Button';
+import FullScreenLoadingIndicator from '@components/FullscreenLoadingIndicator';
+import Modal from '@components/Modal';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import getQuickBooksOnlineSetupLink from '@libs/actions/connections/getQuickBooksOnlineSetupLink';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {ConnectToQuickbooksOnlineButtonOnyxProps, ConnectToQuickbooksOnlineButtonProps} from './types';
+
+type WebViewNavigationEvent = WebViewNavigation;
+
+function ConnectToQuickbooksOnlineButton({policyID, session}: ConnectToQuickbooksOnlineButtonProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const webViewRef = useRef<WebView>(null);
+    const [isWebViewOpen, setWebViewOpen] = useState<boolean>(false);
+
+    const authToken = session?.authToken ?? null;
+
+    const renderLoading = () => <FullScreenLoadingIndicator />;
+
+    /**
+     * Handles in-app navigation for webview links
+     */
+    const handleNavigationStateChange = useCallback(({url}: WebViewNavigationEvent) => !!url, []);
+
+    return (
+        <>
+            <Button
+                onPress={() => setWebViewOpen(true)}
```

Comment:
is this going to open the WebView? Have you tested if the web view opens correctly or not?

Reply:
Yes the webView opens correctly - although the link returns a 404 Page. I asked for more depth at that point: https://github.com/Expensify/App/pull/39652#issuecomment-2039640794

Reply:
There is [another PR](https://github.com/Expensify/App/pull/39219) that implements the QBO authentication flow on web view. I just wanted to confirm how much you made progress on this so that I can inform the author of the other PR 😄